The error from the script 

```
2023-07-25T21:24:16.0223620Z v18.13.0
2023-07-25T21:24:18.5656830Z 
2023-07-25T21:24:18.5657790Z changed 18 packages in 2s
2023-07-25T21:24:18.5658630Z 
2023-07-25T21:24:18.5660940Z 28 packages are looking for funding
2023-07-25T21:24:18.5661620Z   run `npm fund` for details
2023-07-25T21:24:18.6089290Z v18.13.0
2023-07-25T21:24:26.9582050Z npm WARN deprecated typedoc-plugin-resolve-crossmodule-references@0.3.3: Upgrade to typedoc >= 0.24 and remove typedoc-plugin-resolve-crossmodule-references from your dependencies
2023-07-25T21:24:39.5363860Z 
2023-07-25T21:24:39.5365200Z changed 555 packages in 21s
2023-07-25T21:24:39.5365670Z 
2023-07-25T21:24:39.5366820Z 57 packages are looking for funding
2023-07-25T21:24:39.5374940Z   run `npm fund` for details
2023-07-25T21:24:48.1640920Z WARN Appium Appium encountered 4 warnings while validating drivers found in manifest /Users/builder/.appium/node_modules/.cache/appium/extensions.yaml
2023-07-25T21:24:48.1642260Z WARN Appium Driver "mac2" has 1 potential problem: 
2023-07-25T21:24:48.1643590Z WARN Appium   - Driver "mac2" (package `appium-mac2-driver`) may be incompatible with the current version of Appium (v2.0.0-beta.64) due to its peer dependency on older Appium v^2.0.0-beta.66. Please upgrade `appium-mac2-driver` to v1.6.3 or newer.
2023-07-25T21:24:48.2132380Z 2.0.0-beta.64
2023-07-25T21:24:56.7748630Z WARN Appium Appium encountered 4 warnings while validating drivers found in manifest /Users/builder/.appium/node_modules/.cache/appium/extensions.yaml
2023-07-25T21:24:56.7749660Z WARN Appium Driver "mac2" has 1 potential problem: 
2023-07-25T21:24:56.7750880Z WARN Appium   - Driver "mac2" (package `appium-mac2-driver`) may be incompatible with the current version of Appium (v2.0.0-beta.64) due to its peer dependency on older Appium v^2.0.0-beta.66. Please upgrade `appium-mac2-driver` to v1.6.3 or newer.
2023-07-25T21:25:05.3720820Z WARN Appium Appium encountered 4 warnings while validating drivers found in manifest /Users/builder/.appium/node_modules/.cache/appium/extensions.yaml
2023-07-25T21:25:05.3722180Z WARN Appium Driver "mac2" has 1 potential problem: 
2023-07-25T21:25:05.3723390Z WARN Appium   - Driver "mac2" (package `appium-mac2-driver`) may be incompatible with the current version of Appium (v2.0.0-beta.64) due to its peer dependency on older Appium v^2.0.0-beta.66. Please upgrade `appium-mac2-driver` to v1.6.3 or newer.
2023-07-25T21:27:05.5190700Z [31mError: npm command 'uninstall appium-windows-driver' failed with code null.
2023-07-25T21:27:05.5191590Z 
2023-07-25T21:27:05.5192220Z STDOUT:
2023-07-25T21:27:05.5192500Z 
2023-07-25T21:27:05.5192770Z 
2023-07-25T21:27:05.5193470Z STDERR:
2023-07-25T21:27:05.5196960Z [39m
2023-07-25T21:27:14.1173640Z WARN Appium Appium encountered 4 warnings while validating drivers found in manifest /Users/builder/.appium/node_modules/.cache/appium/extensions.yaml
2023-07-25T21:27:14.1175030Z WARN Appium Driver "mac2" has 1 potential problem: 
2023-07-25T21:27:14.1176410Z WARN Appium   - Driver "mac2" (package `appium-mac2-driver`) may be incompatible with the current version of Appium (v2.0.0-beta.64) due to its peer dependency on older Appium v^2.0.0-beta.66. Please upgrade `appium-mac2-driver` to v1.6.3 or newer.
2023-07-25T21:27:14.1722830Z - Installing 'appium-windows-driver@2.7.2'
2023-07-25T21:29:14.2172750Z ✖ Installing 'appium-windows-driver@2.7.2'
2023-07-25T21:29:14.2181730Z [31mError: ✖ Encountered an error when installing package: npm command 'install --save-dev --omit=peer --save-exact --global-style --no-package-lock appium-windows-driver@2.7.2 --json' failed with code null.
2023-07-25T21:29:14.2182560Z 
2023-07-25T21:29:14.2183430Z STDOUT:
2023-07-25T21:29:14.2183770Z 
2023-07-25T21:29:14.2184050Z 
2023-07-25T21:29:14.2184920Z STDERR:
2023-07-25T21:29:14.2186070Z [39m
2023-07-25T21:29:22.7763250Z WARN Appium Appium encountered 4 warnings while validating drivers found in manifest /Users/builder/.appium/node_modules/.cache/appium/extensions.yaml
2023-07-25T21:29:22.7764400Z WARN Appium Driver "mac2" has 1 potential problem: 
2023-07-25T21:29:22.7765630Z WARN Appium   - Driver "mac2" (package `appium-mac2-driver`) may be incompatible with the current version of Appium (v2.0.0-beta.64) due to its peer dependency on older Appium v^2.0.0-beta.66. Please upgrade `appium-mac2-driver` to v1.6.3 or newer.
2023-07-25T21:31:22.9123380Z [31mError: npm command 'uninstall appium-uiautomator2-driver' failed with code null.
2023-07-25T21:31:22.9124150Z 
2023-07-25T21:31:22.9125940Z STDOUT:
2023-07-25T21:31:22.9126420Z 
2023-07-25T21:31:22.9126730Z 
2023-07-25T21:31:22.9127520Z STDERR:
2023-07-25T21:31:22.9128780Z [39m
2023-07-25T21:31:31.4787420Z WARN Appium Appium encountered 4 warnings while validating drivers found in manifest /Users/builder/.appium/node_modules/.cache/appium/extensions.yaml
2023-07-25T21:31:31.4789170Z WARN Appium Driver "mac2" has 1 potential problem: 
2023-07-25T21:31:31.4790460Z WARN Appium   - Driver "mac2" (package `appium-mac2-driver`) may be incompatible with the current version of Appium (v2.0.0-beta.64) due to its peer dependency on older Appium v^2.0.0-beta.66. Please upgrade `appium-mac2-driver` to v1.6.3 or newer.
2023-07-25T21:31:31.5319110Z [31mError: ✖ A driver named "uiautomator2" is already installed. Did you mean to update? Run "appium driver update". See installed drivers with "appium driver list --installed".[39m
2023-07-25T21:31:40.1053630Z WARN Appium Appium encountered 4 warnings while validating drivers found in manifest /Users/builder/.appium/node_modules/.cache/appium/extensions.yaml
2023-07-25T21:31:40.1055040Z WARN Appium Driver "mac2" has 1 potential problem: 
2023-07-25T21:31:40.1056600Z WARN Appium   - Driver "mac2" (package `appium-mac2-driver`) may be incompatible with the current version of Appium (v2.0.0-beta.64) due to its peer dependency on older Appium v^2.0.0-beta.66. Please upgrade `appium-mac2-driver` to v1.6.3 or newer.
2023-07-25T21:33:40.2452100Z [31mError: npm command 'uninstall appium-xcuitest-driver' failed with code null.
2023-07-25T21:33:40.2452870Z 
2023-07-25T21:33:40.2453530Z STDOUT:
2023-07-25T21:33:40.2453850Z 
2023-07-25T21:33:40.2454120Z 
2023-07-25T21:33:40.2454720Z STDERR:
2023-07-25T21:33:40.2455410Z [39m
2023-07-25T21:33:48.8584300Z WARN Appium Appium encountered 4 warnings while validating drivers found in manifest /Users/builder/.appium/node_modules/.cache/appium/extensions.yaml
2023-07-25T21:33:48.8585370Z WARN Appium Driver "mac2" has 1 potential problem: 
2023-07-25T21:33:48.8586580Z WARN Appium   - Driver "mac2" (package `appium-mac2-driver`) may be incompatible with the current version of Appium (v2.0.0-beta.64) due to its peer dependency on older Appium v^2.0.0-beta.66. Please upgrade `appium-mac2-driver` to v1.6.3 or newer.
2023-07-25T21:33:48.9115870Z [31mError: ✖ A driver named "xcuitest" is already installed. Did you mean to update? Run "appium driver update". See installed drivers with "appium driver list --installed".[39m
2023-07-25T21:33:57.4403020Z WARN Appium Appium encountered 4 warnings while validating drivers found in manifest /Users/builder/.appium/node_modules/.cache/appium/extensions.yaml
2023-07-25T21:33:57.4404480Z WARN Appium Driver "mac2" has 1 potential problem: 
2023-07-25T21:33:57.4405650Z WARN Appium   - Driver "mac2" (package `appium-mac2-driver`) may be incompatible with the current version of Appium (v2.0.0-beta.64) due to its peer dependency on older Appium v^2.0.0-beta.66. Please upgrade `appium-mac2-driver` to v1.6.3 or newer.
2023-07-25T21:35:57.5561870Z [31mError: npm command 'uninstall appium-mac2-driver' failed with code null.
2023-07-25T21:35:57.5562630Z 
2023-07-25T21:35:57.5564590Z STDOUT:
2023-07-25T21:35:57.5564860Z 
2023-07-25T21:35:57.5565090Z 
2023-07-25T21:35:57.5565590Z STDERR:
2023-07-25T21:35:57.5566220Z [39m
2023-07-25T21:36:06.1438260Z WARN Appium Appium encountered 4 warnings while validating drivers found in manifest /Users/builder/.appium/node_modules/.cache/appium/extensions.yaml
2023-07-25T21:36:06.1439300Z WARN Appium Driver "mac2" has 1 potential problem: 
2023-07-25T21:36:06.1440460Z WARN Appium   - Driver "mac2" (package `appium-mac2-driver`) may be incompatible with the current version of Appium (v2.0.0-beta.64) due to its peer dependency on older Appium v^2.0.0-beta.66. Please upgrade `appium-mac2-driver` to v1.6.3 or newer.
2023-07-25T21:36:06.1971150Z [31mError: ✖ A driver named "mac2" is already installed. Did you mean to update? Run "appium driver update". See installed drivers with "appium driver list --installed".[39m
2023-07-25T21:36:06.3551350Z ##[error]PowerShell exited with code '1'.
2023-07-25T21:36:06.3561910Z ##[section]Finishing: Install Appium
```

--- a/eng/pipelines/common/ui-tests-steps.yml
+++ b/eng/pipelines/common/ui-tests-steps.yml
@@ -39,7 +39,7 @@ steps:
   - pwsh: ./eng/scripts/appium-install.ps1
     displayName: "Install Appium"
     continueOnError: false
-    retryCountOnTaskFailure: 2
+    retryCountOnTaskFailure: 1
 
   - pwsh: ./build.ps1 --target=dotnet --configuration="${{ parameters.configuration }}" --verbosity=diagnostic
     displayName: 'Install .NET'

